Backport adapted to version 4.3.1. Fixes the build with clang 6.0:

    /wrkdirs/usr/ports/net-p2p/ktorrent/work/ktorrent-4.3.1/libktcore/torrent/chunkbarrenderer.cpp:68:16: error: non-constant-expression cannot be narrowed from type 'bt::Uint32' (aka 'unsigned int') to 'int' in initializer list [-Wc++11-narrowing]

From 48622603e48bfc51d7ae284a6dab18e853db61c7 Mon Sep 17 00:00:00 2001
From: Aleix Pol <aleixpol@kde.org>
Date: Wed, 17 Feb 2016 00:16:08 +0100
Subject: [PATCH] Fix build with clang

Reduce implicit type conversions.

REVIEW: 127087
--- libktcore/torrent/chunkbarrenderer.cpp
+++ libktcore/torrent/chunkbarrenderer.cpp
@@ -28,7 +28,7 @@ namespace kt
 {
 	struct Range
 	{
-		int first,last;
+		Uint32 first,last;
 		int fac;
 	};
 
@@ -71,7 +71,7 @@ namespace kt
 			else
 			{
 				Range & l = rs.last();
-				if (l.last == int(i - 1))
+				if (l.last == i - 1)
 				{
 					l.last = i;
 				}
@@ -120,7 +120,7 @@ namespace kt
 			else
 			{
 				Range & l = rs.last();
-				if (l.last == int(i - 1) && l.fac == fac)
+				if (l.last == i - 1 && l.fac == fac)
 				{
 					l.last = i;
 				}
